pyccel/codegen/compiling/compilers.py

```diff
@@ -363,7 +389,7 @@ def compile_shared_library(self, compile_obj, output_folder, verbose = False, sh
         file_out = os.path.join(compile_obj.source_folder, sharedlib_modname+ext_suffix)
 
         cmd = [exec_cmd, *flags, *includes, *libdirs_flags, *linker_libdirs_flags,
-                *m_code, compile_obj.module_target,
+                compile_obj.module_target, *m_code,
```

Comment:
Static libraries are saved in the `m_code` variable. Libraries need to be placed in order such that any symbols are defined after they are used. Otherwise some compilers will not collect the symbols that it doesn't think are needed. So any static libraries need to come after `compile_obj.module_target`